node/pkg/watchers/sui/watcher.go

```diff
@@ -393,35 +362,12 @@ func (e *Watcher) Run(ctx context.Context) error {
 				return ctx.Err()
 
 			case <-timer.C:
-				resp, err := http.Post(e.suiRPC, "application/json", strings.NewReader(`{"jsonrpc":"2.0", "id": 1, "method": "sui_getLatestCheckpointSequenceNumber", "params": []}`)) //nolint:noctx // TODO FIXME we should propagate context with Deadline here.
-				if err != nil {
-					logger.Error("sui_getLatestCheckpointSequenceNumber failed", zap.Error(err))
-					p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDSui, 1)
-					return fmt.Errorf("sui_getLatestCheckpointSequenceNumber failed to post: %w", err)
-				}
-				body, err := io.ReadAll(resp.Body)
-				if err != nil {
-					logger.Error("sui_getLatestCheckpointSequenceNumber failed", zap.Error(err))
-					p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDSui, 1)
-					return fmt.Errorf("sui_getLatestCheckpointSequenceNumber failed to read: %w", err)
-				}
-				resp.Body.Close()
-				logger.Debug("Body before unmarshalling", zap.String("body", string(body)))
-
-				var res SuiCheckpointSN
-				err = json.Unmarshal(body, &res)
-				if err != nil {
-					logger.Error("unmarshal failed into uint64", zap.String("body", string(body)), zap.Error(err))
-					p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDSui, 1)
-					return fmt.Errorf("sui_getLatestCheckpointSequenceNumber failed to unmarshal body: %s, error: %w", string(body), err)
-				}
-
-				height, pErr := strconv.ParseInt(res.Result, 0, 64)
+				height, pErr := e.getLatestCheckpointSN(logger)
```

Comment:
`pErr` can probably just be `err` now since there's only one :slightly_smiling_face:

Reply:
Changed